src/main/java/com/example/farmfarm/Controller/ProductController.java

```diff
@@ -1,4 +1,83 @@
 package com.example.farmfarm.Controller;
 
+import com.example.farmfarm.Entity.ProductEntity;
+import com.example.farmfarm.Service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.util.List;
+
+@Controller
+@RequestMapping("/product")
 public class ProductController {
+    @Autowired
+    private ProductService productService;
+
+    // 상품 등록 Form
+    @GetMapping("/new")
+    public ModelAndView getProductForm(HttpServletRequest req, @ModelAttribute("product")ProductEntity product) {
+        HttpSession session = req.getSession(false);
+        Long f_id = (long)session.getAttribute("f_id");
+        ModelAndView mav = new ModelAndView("");    // 추후에 이동할 view 이름 지정하기
+        mav.addObject("farm_id", f_id);
+        return mav;
+    }
+
+    // 상품 등록
+    @ResponseBody
+    @PostMapping()
+    public ResponseEntity<Object> registerProduct(HttpServletRequest request, @RequestBody ProductEntity product) {
+        ProductEntity newProduct = productService.saveProduct(product);
+        return ResponseEntity.ok().body(newProduct);
+    }
+
+    // 상품 조회
+    @GetMapping("{/p_id}")
+    public ResponseEntity<Object> getProduct(@PathVariable("p_id") long p_id) {
+        ProductEntity product = productService.getProduct(p_id);
+        return ResponseEntity.ok().body(product);
+    }
+
+    // 상품 리스트 조회
+    @GetMapping()
+    public ResponseEntity<Object> getAllProduct(HttpServletRequest request){
+        List productList = productService.getAllProduct();
+        return ResponseEntity.ok().body(productList);
+    }
```

Comment:
이 부분은 상품 리스트를 가져오는 부분이기 때문에 반환타입이 List 로 변경되어야할 것 같아요 !! 
맞나요 ?  @hyeonahhh  

```suggestion
public List<ProductEntity> getAllProduct(HttpServletRequest request){
        List<ProductEntity> productList = productService.getAllProduct();
        return productList;
    }

```

Reply:
ReponseEntity<Object>로 반환타입 줘도 될 것 같아요~!!